crates/driver/src/infra/simulator/enso/mod.rs

```diff
@@ -8,23 +14,48 @@ const GAS_LIMIT: u64 = 30_000_000;
 pub(super) struct Enso {
     url: reqwest::Url,
     chain_id: eth::ChainId,
+    current_block: CurrentBlockStream,
+    network_block_interval: Option<Duration>,
 }
 
 #[derive(Debug, Clone)]
 pub struct Config {
     /// The URL of the Transaction Simulator API.
     pub url: reqwest::Url,
+    /// The time between new blocks in the network.
+    pub network_block_interval: Option<Duration>,
 }
 
 impl Enso {
-    pub(super) fn new(config: Config, chain_id: eth::ChainId) -> Self {
+    pub(super) fn new(
+        config: Config,
+        chain_id: eth::ChainId,
+        current_block: CurrentBlockStream,
+    ) -> Self {
         Self {
             url: reqwest::Url::parse(&format!("{}api/v1/simulate", config.url)).unwrap(),
             chain_id,
+            current_block,
+            network_block_interval: config.network_block_interval,
         }
     }
 
     pub(super) async fn simulate(&self, tx: eth::Tx) -> Result<eth::Gas, Error> {
+        let current_block = *self.current_block.borrow();
+
+        let (block_number, block_timestamp) = match self.network_block_interval {
+            None => (None, None), // use default values which result in simulation on `latest`
+            Some(duration) => {
+                // We would like to simulate on the `pending` block instead of the `latest`
+                // block. Unfortunately `enso` does not support that so to get closer to
+                // the actual behavior of the `pending` block we use the block number of
+                // the `latest` block but the timestamp of the `pending` block.
```

Comment:
Could we technically simulate a bundle where the first transaction does a trivial operation on the latest block and the second transaction is the actual simulation on current + 1?

Reply:
Yeah, I think this would almost be identical to simulating on `pending` (assuming other simulators already compute the correct `basefee` for `pending`).
I'd just like to do it in a separate PR and after the required PR was merged into `temper`.

Reply:
The enso team is fine with simulating on a future block so I will open a PR in `temper` so we don't have to resort to those workarounds in our code base.